algebird-core/src/main/scala/com/twitter/algebird/MapAlgebra.scala

```diff
@@ -222,4 +224,31 @@ object MapAlgebra {
 
   def dot[K, V](left: Map[K, V], right: Map[K, V])(implicit mring: Ring[Map[K, V]], mon: Monoid[V]): V =
     Monoid.sum(mring.times(left, right).values)
+
+  def cube[K, V](it: TraversableOnce[(K, V)])(implicit c: Cuber[K]): Map[c.K, TraversableOnce[V]] = {
+    val sg = new Semigroup[Iterable[V]] {
+      def plus(x: Iterable[V], y: Iterable[V]) = x ++ y
+    }
+    cubeSum(it.map { case (k, v) => (k, Iterable(v)) })(c, sg)
```

Comment:
I can't seem to find an immutable wrapper to a mutable map in the standard library. So our options here are 

* write a wrapper class
* return an immutable map by calling `.toMap` on the map
* return the mutable map, but specify the return type as `scala.collection.Map` instead of `scala.collection.mutable.Map`. In this case, the user doesn't know that it's mutable, so it's effectively immutable. (They can only access the mutator methods by casting.) If they really want a `immutable.Map`, they can call `.toMap` on it.